libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt

```diff
@@ -168,6 +178,67 @@ class Poller(private val configFactory: ConfigFactoryProtocol, debounceTimer: Ti
         }
     }
 
+    private fun poll(userProfileOnly: Boolean, snode: Snode, deferred: Deferred<Unit, Exception>): Promise<Unit, Exception> {
+        if (userProfileOnly) {
+            return pollUserProfile(snode, deferred)
+        }
+        return poll(snode, deferred)
+    }
+
+    private fun pollUserProfile(snode: Snode, deferred: Deferred<Unit, Exception>): Promise<Unit, Exception> {
+        return task {
+            runBlocking(Dispatchers.IO) {
+                val requests = mutableListOf<SnodeAPI.SnodeBatchRequestInfo>()
+                val hashesToExtend = mutableSetOf<String>()
+                configFactory.user?.let { config ->
+                    hashesToExtend += config.currentHashes()
+                    SnodeAPI.buildAuthenticatedRetrieveBatchRequest(
+                            snode, userPublicKey,
+                            config.configNamespace(),
+                            maxSize = -8
+                    )
+                }?.let { request ->
+                    requests += request
+                }
+
+                if (hashesToExtend.isNotEmpty()) {
+                    SnodeAPI.buildAuthenticatedAlterTtlBatchRequest(
+                            messageHashes = hashesToExtend.toList(),
+                            publicKey = userPublicKey,
+                            newExpiry = SnodeAPI.nowWithOffset + 14.days.inWholeMilliseconds,
+                            extend = true
+                    )?.let { extensionRequest ->
+                        requests += extensionRequest
+                    }
+                }
+
+                SnodeAPI.getRawBatchResponse(snode, userPublicKey, requests).bind { rawResponses ->
+                    isCaughtUp = true
+                    if (deferred.promise.isDone()) {
+                        return@bind Promise.ofSuccess(Unit)
```

Comment:
These labels are not the nicest to read...

Do you actually need bind, or can you just call ` success {` ?

Reply:
I don't like those either, let me see what I can do...

Reply:
I managed to remove the labels in this function but I noticed there are a lot more in other functions. We might need a bit cleaning up on them as well.